src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -155,6 +180,14 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
           log.info("Writing new beam data to[%s]: %s", dataPath, new String(newMetaBytes))
           curator.setData().forPath(dataPath, newMetaBytes)
         }
+        zkMetadataCache = newMeta
+
+        log.debug(zkMetadataCache.latestCloseTime.toString)
```

Comment:
Given that you are doing iteration here only to log debug, it's probably best to wrap this in a `if (log.isDebugEnabled())`